packages/react-app/src/contexts/BridgeContext.jsx

```diff
@@ -148,11 +154,41 @@ export const BridgeProvider = ({ children }) => {
     [setToken, fromToken, toToken],
   );
 
+  const checkForCustomToken = useCallback(async () => {
+    const { token: address, status } = isTokenInjected;
+    if (!status) {
```

Comment:
This is calling setToken even if token === null, we should have a check for that.